sonic_platform_base/module_base.py

```diff
@@ -68,6 +89,83 @@ def get_system_eeprom_info(self):
         """
         raise NotImplementedError
 
+    def get_name(self):
+        """
+        Retrieves the name of the module prefixed by SUPERVISOR, LINE-CARD,
+        FABRIC-CARD
+
+        Returns:
+            string: A string providing the name of the card prefixed by one of the
+            MODULE_TYPE_SUPERVISOR, MODULE_TYPE_LINE, MODULE_TYPE_FABRIC followed by
+            an index.
+            Ex. A Chassis having 1 supervisor, 4 line-cards and 6 fabric-cards
+            can provide names SUPERVISOR0, LINE-CARD0 to LINE-CARD3,
+            FABRIC-CARD0 to FABRIC-CARD5
+        """
+        raise NotImplementedError
+
+    def get_description(self):
+        """
+        Retrieves the platform vendor's product description of the module
+
+        Returns:
+            string: A string providing the product description of the module. This
+            is vendor specific.
+        """
+        raise NotImplementedError
+
+    def get_slot(self):
+        """
+        Retrieves the platform vendor's slot number of the module
+
+        Returns:
+            An integer, indicating the slot number in the chassis
+        """
+        raise NotImplementedError
+
+    def get_type(self):
+        """
+        Retrieves the type of the module.
+
+        Returns:
+            string: A string providing the module-type. Supported values are
+            MODULE_TYPE_SUPERVISOR, MODULE_TYPE_LINE, MODULE_TYPE_FABRIC
+        """
+        raise NotImplementedError
+
+    def get_status(self):
+        """
+        Retrieves the status of the card
+
+        Returns:
+            string: A string providing the status of the module. Support values
+            are MODULE_STATUS_EMPTY, MODULE_STATUS_OFFLINE, MODULE_STATUS_FAULT,
+            MODULE_STATUS_PRESENT, MODULE_STATUS_ONLINE
+        """
+        raise NotImplementedError
+
+    def reboot(self):
+        """
+        Request to reboot the module
+
+        Returns:
+            bool: True if the request has been issued successfully, False if not
+        """
+        raise NotImplementedError
+
+    def set_admin_state(self, up):
+        """
+        Request to keep the card in administratively up/down state.
+        The down state will power down the module and the status should show
+        MODULE_STATUS_OFFLINE.
+        The up state will take the module to MODULE_STATUS_PRESENT,
+        MODULE_STATUS_FAULT or MODULE_STAUS_ONLINE states.
+
+        Returns:
+            bool: True if the request has been issued successfully, False if not
+        """
+        raise NotImplementedError
```

Comment:
Can you add an "Args" section of this docstring to describe the type of value expected to be passed in as `up`?

Reply:
Fixed.